### ISSUE TITLE
[FEATURE REQ] DefaultAzureCredential for local docker testing

### ISSUE THREAD

**Issue by karpikpl** [microsoft/vscode-docker#3871]
Recently Visual Studio added support for `VisualStudioCredential` to support local docker testing.
https://github.com/Azure/azure-sdk-for-net/issues/19167#issuecomment-1439472328

Is there a plan to support similar feature for VSCode?

Downside of Visual Studio solution is that it requires Docker Desktop and doesn't work on WSL.
Sharing AZ CLI credentials is complicated and bloats the image size by over 1 GB - all those issues are described in the linked issue.

Ideally there's a way to pass credentials same way Visual Studio is doing?

**Comment by dbreshears** [comment]
Agreed, let's investigate to see what can be done here to share this.

**Comment by AndreiCalin24** [comment]
Hi guys! Is there any update on this? Or workarounds?
